common/djangoapps/enrollment/data.py

```
@@ -107,27 +108,30 @@ def create_course_enrollment(username, course_id, mode, is_active):
-        log.warn(msg)
-        raise UserNotFoundError(msg)
-
-    try:
-        enrollment = CourseEnrollment.enroll(user, course_key, check_access=True)
-        return _update_enrollment(enrollment, is_active=is_active, mode=mode)
-    except NonExistentCourseError as err:
-        raise CourseNotFoundError(err.message)
-    except EnrollmentClosedError as err:
-        raise CourseEnrollmentClosedError(err.message)
-    except CourseFullError as err:
-        raise CourseEnrollmentFullError(err.message)
-    except AlreadyEnrolledError as err:
-        enrollment = get_course_enrollment(username, course_id)
-        raise CourseEnrollmentExistsError(err.message, enrollment)
-
+    current_course_org = course_id.split(':')[1].split('+')[0]
+    site_course_org_filter = configuration_helpers.get_value('course_org_filter')
+    if site_course_org_filter == site_course_org_filter or not site_course_org_filter:
+        try:
+            user = User.objects.get(username=username)
+        except User.DoesNotExist:
+            msg = u"Not user with username '{username}' found.".format(username=username)
+            log.warn(msg)
+            raise UserNotFoundError(msg)
+
+        try:
+            enrollment = CourseEnrollment.enroll(user, course_key, check_access=True)
+            return _update_enrollment(enrollment, is_active=is_active, mode=mode)
+        except NonExistentCourseError as err:
+            raise CourseNotFoundError(err.message)
+        except EnrollmentClosedError as err:
+            raise CourseEnrollmentClosedError(err.message)
+        except CourseFullError as err:
+            raise CourseEnrollmentFullError(err.message)
+        except AlreadyEnrolledError as err:
+            enrollment = get_course_enrollment(username, course_id)
+            raise CourseEnrollmentExistsError(err.message, enrollment)
+    else:
+        return NonExistentCourseError
```

unrelated: this is actually how i wanted the original fix to be written. in other words, i'd rather throw this error than conditionalizing some markup in the view. we should consider adopting a similar strategy when we revisit this area soon